news: add newsfile entries for PRs before the newsfile generating script

alltilla opened the pull request:
If I added you to the reviewers, that is because I wrote the news entry about the PR you have created. Please leave comment if you would have wrote something differently.

Signed-off-by: Attila Szakacs <attila.szakacs@oneidentity.com>

kira-syslogng commented:
Build FAILURE

alltilla commented:
@kira-syslogng retest this please

kira-syslogng commented:
Build SUCCESS

gaborznagy commented on news/feature-3050.md:
```diff
@@ -0,0 +1 @@
+`rewrite`: Added type hinting support.
```
I would say the PR introduces the possibility to use template options, but I don't think there is any type-hint for a template. I would also say that this is a bugfix: I think template options should be usable everywhere we a template is accepted.

About type-hint:
I've checked the code, I see that there is some intention to support this, but on LogTemplate side there is nothing about typehints, except that it's an member of LogTemplate.

alltilla replied:
@Kokan I might have misunderstood this, can you double-check please? 10Q!

Kokan replied:
Not it was my mistake, only concluded that from the grammar intent. But as @gaborznagy explained the user must act upon, which template does not do.

alltilla replied:
So if I understand right, this is just a refactor? I removed the news entry, but I am happy to put it back, if needed.

Kokan replied:
Yes, just refactor.

alltilla commented on news/developer-note-3067.md:
```diff
@@ -0,0 +1 @@
+`cmake`: Now we install `loggen` headers, as we do with `autotools`.
```
@lbudai What is your opinion? Asking you, as you wrote this code.

alltilla replied:
Kinda fits in bugfix for me, too, I moved it.

lbudai replied:
I wouldn't say this as a bugfix...  it's just a 'small fix' and it was intentional that I did not add any notes on this.
(I added news file fragment for all my PRs I wanted to be mentioned in the news file. But if you think it should be added to the news file, you can do that and I'm okay with this description.)

alltilla replied:
Thanks, I let it be there still.

alltilla commented on news/developer-note-3079.md:
```diff
@@ -0,0 +1 @@
+`commit-check`: Now we allow whitespace in commit messages.
```
However I think otherwise, I believe it is not a mandatory entry in the NEWS file, if no one fights for it I will remove it :)

alltilla replied:
Removed.

kira-syslogng commented:
Build SUCCESS

lbudai commented on news/bugfix-3064.md:
```diff
@@ -0,0 +1 @@
+`loggen`: Fixed message loss with TLSv1.3.
```
I think we should add a more detailed description related to this issue.

MrAnno replied:
@alltilla I've opened a PR in your fork, it contains a longer description for this bugfix.
https://github.com/alltilla/syslog-ng/pull/5/files

alltilla replied:
Thanks, merged :)

lbudai commented on news/feature-3051.md:
```diff
@@ -0,0 +1 @@
+`geoip2`: Added `template()` option as an alias for the positinal argument string, to match the grammar convention.
```
typo

lbudai commented on news/other-2979.md:
```diff
@@ -0,0 +1 @@
+`scl`: Moved `scl` files to the core package.
```
This is a packaging/debian related change, please add this information to the description.

lbudai replied:
(I think we should add a dedicated news file category for packaging.)

alltilla replied:
Good suggestion, let's bring this up at the retro :) Thanks!

lbudai commented on news/developer-note-3038.md:
```diff
@@ -0,0 +1 @@
+`dbdl`: Added option to customize `shell` command.
```
typo

lbudai commented on news/bugfix-3056.md:
```diff
@@ -0,0 +1 @@
+`checkpoint`: Fixed parsing ISO timestamp.
```
checkpoint parser

kira-syslogng commented:
Build SUCCESS

kira-syslogng commented:
Build SUCCESS

alltilla commented:
@bazsi: Hi Bazsi!

Could you check the news entries related to your changes?

```
$ cat news/feature-3015.md
`scl`: Improved error message at init, when an `scl` is missing a dependency.

$ cat news/other-2979.md
`packaging`: Moved `scl` files to the core package.

$ cat news/other-3072.md
doc: Added manual page for `persist-tool`.

```

Thanks! :)

lbudai commented:
@bazsi : as we have a depending PR (#3151), and that need to be rebased, I'm merge this as-is.
If there is something you would like to change, you could still open a PR before the release.
